packages/flutter_tools/templates/module/ios/library/Flutter.tmpl/podhelper.rb

```diff
@@ -1,75 +1,112 @@
-if !File.exist?(engine_dir)
+  engine_dir = File.join(__dir__, 'engine')
+  if !File.exist?(engine_dir)
     # Copy the debug engine to have something to link against if the xcode backend script has not run yet.
-    debug_framework_dir = File.join(flutter_root(flutter_application_path), 'bin', 'cache', 'artifacts', 'engine', 'ios')
+    # CocoaPods will not generate a install framework command if the vendored framework does not exist.
+    debug_framework_dir = File.join(flutter_root, 'bin', 'cache', 'artifacts', 'engine', 'ios')
     FileUtils.mkdir_p(engine_dir)
     FileUtils.cp_r(File.join(debug_framework_dir, 'Flutter.framework'), engine_dir)
     FileUtils.cp(File.join(debug_framework_dir, 'Flutter.podspec'), engine_dir)
-end
+  end
 
-pod 'Flutter', :path => engine_dir
-pod 'FlutterPluginRegistrant', :path => File.join(framework_dir, 'FlutterPluginRegistrant')
+  pod 'Flutter', :path => engine_dir, :inhibit_warnings => true
+  pod 'FlutterPluginRegistrant', :path => File.join(__dir__, 'FlutterPluginRegistrant'), :inhibit_warnings => true
 
-symlinks_dir = File.join(framework_dir, '.symlinks')
-FileUtils.mkdir_p(symlinks_dir)
-plugin_pods = parse_KV_file(File.join(flutter_application_path, '.flutter-plugins'))
-plugin_pods.map { |r|
+  symlinks_dir = File.join(__dir__, '.symlinks')
+  FileUtils.mkdir_p(symlinks_dir)
+  plugin_pods = parse_KV_file(File.join(flutter_application_path, '.flutter-plugins'))
+  plugin_pods.map do |r|
     symlink = File.join(symlinks_dir, r[:name])
     FileUtils.rm_f(symlink)
     File.symlink(r[:path], symlink)
-    pod r[:name], :path => File.join(symlink, 'ios')
-}
+    pod r[:name], :path => File.join(symlink, 'ios'), :inhibit_warnings => true
+  end
+end
+
+# Install pod needed to embed Flutter application.
+def install_flutter_application_pod(flutter_application_path, flutter_application_name = nil)
+  flutter_application_name ||= File.basename(flutter_application_path)
+
+  app_framework_dir = File.join(__dir__, 'App.framework')
+  app_framework_dylib = File.join(app_framework_dir, 'App')
+  if !File.exist?(app_framework_dylib)
+    # Fake an App.framework to have something to link against if the xcode backend script has not run yet.
+    # CocoaPods will not generate a install framework command if the vendored framework binary does not exist.
+    FileUtils.mkdir_p(app_framework_dir)
+    `echo "static const int Moo = 88;" | xcrun clang -x c -dynamiclib -o "#{app_framework_dylib}" -`
+  end
+
+  pod flutter_application_name, :path => __dir__, :inhibit_warnings => true
+
+  # Use relative paths for script phase paths to be checked into source control.
+  # Process will be run from project directory.
+  current_directory_pathname = Pathname.new __dir__.to_s
+  project_directory_pathname = Pathname.new Dir.pwd
+  relative = current_directory_pathname.relative_path_from project_directory_pathname
+
+  flutter_build_script_path = File.join('${SRCROOT}', relative, 'flutter_build_phase.sh');
```

Comment:
I assume we'd have to document in the wiki that if your outer project and flutter module are source controlled separately, your other team members who check them out have to let them be in the same relative paths (or create symlinks)

Reply:
The relative links here are being generated as the difference from where the outer host Xcode project is (Dir.pwd) and the current directory of this podhelper.rb file (`__dir__`).
Though your comment is true anyway. The Podfile needs to be checked in, which needs to pass in the path where the flutter module is located, even with the current setup: https://github.com/flutter/flutter/wiki/Add-Flutter-to-existing-apps#add-your-flutter-app-to-your-podfile

Reply:
Oh I see what you mean.  The relative strings will be checked into source control since they become the inputs into the build phase.  I could change this to :input_file_list_paths and pass in a generated xcfilelist, but that feature is only in Xcode 10.

Reply:
Right, the relative-ness of the outer project and the flutter module might not be the same for all the people in a team who check these 2 gits out. We probably can't make everything 100% magical but it would be nice to account for that scenario and minimize the git dirty-ness (possibly by best practice and documentation)